src/commonMain/kotlin/fr/acinq/lightning/channel/InteractiveTx.kt

```diff
@@ -770,17 +828,13 @@ data class InteractiveTxSigningSession(
                 fundingTxIndex = fundingTxIndex, fundingTxHash = unsignedTx.hash, fundingTxOutputIndex = sharedOutputIndex,
                 remoteFundingPubkey = fundingParams.remoteFundingPubkey,
                 remotePerCommitmentPoint = remotePerCommitmentPoint
-            ).flatMap { firstCommitTx ->
+            ).map { firstCommitTx ->
                 val localSigOfRemoteTx = Transactions.sign(firstCommitTx.remoteCommitTx, channelKeys.fundingKey(fundingTxIndex))
                 val commitSig = CommitSig(channelParams.channelId, localSigOfRemoteTx, listOf())
-                when (val signedFundingTx = sharedTx.sign(keyManager, fundingParams, channelParams.localParams)) {
-                    null -> Either.Left(ChannelFundingError(channelParams.channelId))
-                    else -> {
-                        val unsignedLocalCommit = UnsignedLocalCommit(commitmentIndex, firstCommitTx.localSpec, firstCommitTx.localCommitTx, listOf())
-                        val remoteCommit = RemoteCommit(commitmentIndex, firstCommitTx.remoteSpec, firstCommitTx.remoteCommitTx.tx.txid, remotePerCommitmentPoint)
-                        Either.Right(Pair(InteractiveTxSigningSession(fundingParams, fundingTxIndex, signedFundingTx, Either.Left(unsignedLocalCommit), remoteCommit), commitSig))
-                    }
-                }
+                val unsignedLocalCommit = UnsignedLocalCommit(commitmentIndex, firstCommitTx.localSpec, firstCommitTx.localCommitTx, listOf())
+                val remoteCommit = RemoteCommit(commitmentIndex, firstCommitTx.remoteSpec, firstCommitTx.remoteCommitTx.tx.txid, remotePerCommitmentPoint)
+                val signedFundingTx = sharedTx.sign(keyManager, fundingParams, channelParams.localParams)
```

Comment:
Looks much nicer. I guess the idea is that if the signatures don't match an error will be raised by the peer instead of here with the `ChannelFundingError(channelParams.channelId)`?

Reply:
Exactly, we checked that we have the correct number of signatures, and we will verify that the transaction correctly spends its inputs later and fail at that point if one of the signatures was invalid. If we forgot to send some of our signatures, our peer will send back an error.